examples/linked_list.zig

```diff
@@ -0,0 +1,322 @@
+const std = @import("std");
+const zbench = @import("zbench");
+const Benchmark = zbench.Benchmark;
+
+// We can collect runners in a tuple like this to easily iterate over them
+const runners = .{
+    // Runner for appending 128 elements
+    struct {
+        const Self = @This();
+        const name = "linked-list append-128";
+        ll: LinkedList(i128),
+        pub fn init(alloc: std.mem.Allocator) !Self {
+            return Self{ .ll = LinkedList(i128).init(alloc) };
+        }
+        pub fn run(self: *Self) void {
+            for (0..128) |i| self.ll.append(i) catch @panic("Alloc error!");
+        }
+        pub fn deinit(self: *Self) void {
+            self.ll.deinit();
+        }
+    },
+
+    // Runner for popping from list with 128 elements
+    struct {
+        const Self = @This();
+        const name = "linked-list pop-128";
+        ll: LinkedList(i128),
+        pub fn init(alloc: std.mem.Allocator) !Self {
+            var ll = LinkedList(i128).init(alloc);
+            for (0..128) |i| ll.append(i) catch @panic("Alloc error!");
+            return Self{ .ll = ll };
+        }
+        pub fn run(self: *Self) void {
+            for (0..128) |_| _ = self.ll.pop();
+        }
+        pub fn deinit(self: *Self) void {
+            self.ll.deinit();
+        }
+    },
+
+    // Runner for iterating over a list with 128 elements and addint 1 to each element
+    struct {
+        const Self = @This();
+        const name = "linked-list iter-128";
+        ll: LinkedList(i128),
+        pub fn init(alloc: std.mem.Allocator) !Self {
+            var ll = LinkedList(i128).init(alloc);
+            for (0..128) |i| ll.append(i) catch @panic("Alloc error!");
+            return Self{ .ll = ll };
+        }
+        pub fn run(self: *Self) void {
+            var it = self.ll.iterStart();
+            while (it.next()) |elem| {
+                elem.* += 1;
+            }
+        }
+        pub fn deinit(self: *Self) void {
+            self.ll.deinit();
+        }
+    },
+};
+
+pub fn main() !void {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+
+    const runs: usize = 128;
+    var bench = try Benchmark.init(1_000_000_000, runs, gpa.allocator());
```

Comment:
is the benchmark and the "benchmarked" code getting the same allocator? Not sure if I got that correctly ;-) In general I think those should be separate allocators. You might want to benchmark if your code runs faster with another allocator; a linked list is a good example that benefits from an ArenaAllocator IIRC.

Reply:
That is very true, that option should be on the table for every benchmark we create